plugins/inventory/kubevirt.py

```diff
@@ -372,25 +398,18 @@ def setup(self, config_data: Dict, cache: bool, cache_key: str) -> None:
         gets the configured connections and runs fetch_objects on them.
         If there is a cache it is returned instead.
         """
-        connections = config_data.get("connections")
-
         if not HAS_K8S_MODULE_HELPER:
             raise KubeVirtInventoryException(
                 "This module requires the Kubernetes Python client. "
-                + f"Try `pip install kubernetes`. Detail: {k8s_import_exception}"
+                + f"Try `pip install kubernetes`. Detail: {K8S_IMPORT_EXCEPTION}"
             )
 
-        source_data = None
         if cache and cache_key in self._cache:
-            try:
-                source_data = self._cache[cache_key]
-            except KeyError:
-                pass
+            return
 
-        if not source_data:
-            self.fetch_objects(connections)
+        self.fetch_objects(config_data.get("connections", []))
 
-    def fetch_objects(self, connections: Dict) -> None:
+    def fetch_objects(self, connections: List[Dict]) -> None:
```

Comment:
Since we are already passing a list of connection, is it required to check `if not isinstance(connection, list)`?

Reply:
This is only a type hint and is not being enforced. Python is a dynamic language and in this case `connections` is user input and could be anything.